aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -118,15 +118,44 @@ impl<'e, E: ExecutorView> StorageAdapter<'e, E> {
             };
 
             let buf_size = resource_size(&buf);
-            Ok((buf, buf_size + group_size as usize))
-        } else {
-            let state_key = resource_state_key(address, struct_tag)?;
-            let buf = self
-                .executor_view
-                .get_resource_bytes(&state_key, maybe_layout)?;
-            let buf_size = resource_size(&buf);
-            Ok((buf, buf_size))
+            return Ok(Some((buf, buf_size + group_size as usize)));
         }
+        Ok(None)
+    }
+
+    fn get_any_resource_with_layout(
+        &self,
+        address: &AccountAddress,
+        struct_tag: &StructTag,
+        metadata: &[Metadata],
+        maybe_layout: Option<&MoveTypeLayout>,
+    ) -> PartialVMResult<(Option<Bytes>, usize)> {
+        let resource_group_bytes = self.get_resource_group_bytes(address, struct_tag, metadata, maybe_layout)?;
+        if let Some(result) = resource_group_bytes {
+            return Ok(result);
+        }
+
+        let state_key = resource_state_key(address, struct_tag)?;
+        let buf = self
+            .executor_view
+            .get_resource_bytes(&state_key, maybe_layout)?;
+        let buf_size = resource_size(&buf);
+        Ok((buf, buf_size))
+    }
+
+    fn get_any_resource_size_with_layout(
+        &self,
+        address: &AccountAddress,
+        struct_tag: &StructTag,
+        metadata: &[Metadata],
+        maybe_layout: Option<&MoveTypeLayout>,
+    ) -> PartialVMResult<Option<u64>> {
+        let resource_group_bytes = self.get_resource_group_bytes(address, struct_tag, metadata, maybe_layout)?;
+        if let Some((_, size)) = resource_group_bytes {
+            return Ok(Some(size as u64));
```

Comment:
Hmmmm, so this is wrong I think because it is possible that group does not contain member `A`, so here we have `None, some_size` where `some_size` may or may not be zero depending if this is the first access to this group. 

We basically have 3 cases:
1. Resource exists - `Some(size)`, `None` otherwise
2. Resource group member exists - `Some(size /*does not reflect bytes.len()*/)`
3. Resource group member does not exist, `None` but we need to pass some `size` to the VM because if it is a first access, we need to charge group size.

Maybe, we can have an API which returns `Option<u64>` - resource or resuorce member size, and also additional `NumBytes` as "additional charge", so we can handle groups crrectly.

cc @zekun000 @gelash